MapboxNavigation/JunctionView.swift

```diff
@@ -50,18 +51,30 @@ public class JunctionView: UIImageView {
                 imageRepository.imageWithURL(guidanceViewImageURL, cacheKey: guidanceView.cacheKey!) { [unowned self] (downloadedImage) in
                     
                     self.isCurrentlyVisible = true
-                    self.isHidden = !self.isCurrentlyVisible
                     
                     DispatchQueue.main.async {
+                        self.isHidden = !self.isCurrentlyVisible
                         self.image = downloadedImage
-                        self.show(true)
                     }
                 }
             }
         }
         
     }
     
+    public func updateDistance(for currentStepProgress: RouteStepProgress) {
```

Comment:
Add a documentation comment like “Show or hide the junction view based on the distance traveled in an updated step progress.”

Reply:
Will do.